File: api/src/main/java/dev/ulloasp/mlsuite/role/application/dto/RolesResponseDto.java
package dev.ulloasp.mlsuite.role.application.dto;

import java.util.List;

public record RolesResponseDto(
        List<RoleDefinitionDto> roles,
        List<RoleTemplateDto> templates,
        List<PermissionGroupDto> permissionCatalog,
        RoleStatsDto stats) {
}

File: frontend/src/api/workspace/dtos/roles-response-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

import type { PermissionGroupDto, RoleDefinitionDto, RoleTemplateDto } from "./index";

export interface RolesResponseDto {
  roles: RoleDefinitionDto[];
  templates: RoleTemplateDto[];
  permissionCatalog: PermissionGroupDto[];
  stats: {
    customRoles: number;
    lockedRoles: number;
    assignedUsers: number;
  };
}
